athf/core/metrics.py

```
@@ -0,0 +1,576 @@
+                    ", ".join(EVENT_TYPES), self.event_type
+                )
+            )
+
+    def to_dict(self) -> Dict[str, Any]:
+        """Render to a plain dict, dropping ``None`` fields except event_type."""
+        raw = asdict(self)
+        keep = {"event_type", "timestamp", "event_id", "custom"}
+        return {k: v for k, v in raw.items() if k in keep or v is not None}
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "MetricEvent":
```

Ruff (0.15.18)

[warning] Remove quotes from type annotation

Remove quotes

(UP037)